crates/telio-firewall/src/firewall.rs

```diff
@@ -610,6 +656,20 @@ impl StatefullFirewall {
     fn handle_inbound_icmp<'a, P: IpPacket<'a>>(&self, peer: &PublicKey, ip: &P) -> bool {
         let icmp_packet = unwrap_option_or_return!(P::Icmp::new(ip.payload()), false);
 
+        let key = IcmpConn {
+            remote_addr: ip.get_source().into(),
+            local_addr: ip.get_destination().into(),
+            identifier: 0,
+        };
+        let mut icmp_cache = unwrap_lock_or_return!(self.icmp.lock(), false);
+        if icmp_cache.get(&key).is_some() {
+            telio_log_trace!("Matched ICMP conntrack entry {:?}", key,);
+            telio_log_trace!("Removing ICMP conntrack entry {:?}", key);
```

Comment:
I don't think it makes sense now. With the previous logic we allowed all incoming traffic, so basically we've been opening the firewall for all ICMP traffic. If the logic is that we should only allow one response packet, then it only  makes sense to let the packet that really is the response, not any packet. So the only candidate for me that should be placed in the cache is an echo req, and it should be removed by a matching echo response. That's it. With this implementation by sending echo req we allow the other side to send an echo req to us which seems strange.

Reply:
Ok more candidates for the cache: Basically all from BLOCKED_TYPES and ICMPV6_BLOCKED_TYPES, but only those.

Reply:
And also because now we have more packet types in the cache, not only echo, the ICMP type and maybe even code, should also be part of the cache key.

Reply:
And one additional note: The key in cache should probably have repose type. So the logic would go: when we have outbound echo request we put the key in cache with the same code, identifier and seq number and with echo response type. And when this response comes we remove this key. That's it. I've also  checked that all ipv4 req/resp packets that we have in BLOCKED_TYPES have identifier and seq number

Reply:
All good points, and all addressed